### ISSUE TITLE
Design library

### ISSUE THREAD

**Issue by ekolis** [ekolis/stars-nova#55]
**Describe the feature you'd like to see**
A "library" for storing ship/base designs between games (possibly also build queues?) See the implementation in FrEee.

**Benefits**
You don't have to redesign your ships from scratch every game.

**Describe alternatives**
N/A

**Additional context**
N/A

**Comment by marhawkman** [comment]
yeah, I cant think of any good reasons NOT to(other than coding work).

**Comment by ekolis** [comment]
Well, I guess we would need to distinguish between active and inactive designs if we want to keep the design cap - but that's a silly game mechanic that I'd just as well do away with!

**Comment by marhawkman** [comment]
It was a game-ism that was dumb.  AFAIK it was part of the way it saved data in game files.  IE each player file had X design slots in the data structure and that was it.

I never liked it because it felt artificial.  MoO2 actually didn't care about designs.  you have 5 design slots...  BUT, you can built an endless stream of custom one-off ships because each ship has it's own space in the game save for it's component list.  You could build a custom ship, then after queueing construction change the design to make something else.

Not sure I'd go THAT far, SE5's way worked nicely.  But having a  save/load function would be great.  I literally make designs the exact same way as-is.  More so in MoO2 and Stars way since there are LESS options.
